messaginginapp/src/sharedTest/java/io/customer/messaginginapp/GistSdkListenersTest.kt

```diff
@@ -0,0 +1,124 @@
+package io.customer.messaginginapp
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.customer.commontest.BaseTest
+import io.customer.messaginginapp.gist.data.model.Message
+import io.customer.messaginginapp.gist.presentation.GistListener
+import io.customer.messaginginapp.gist.presentation.GistSdk
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.thread
+import org.amshove.kluent.internal.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class GistSdkListenersTest : BaseTest() {
+    @Test
+    fun processAndRemoveListenersIndividually_givenConcurrentModification_expectSuccessfulCompletion() {
```

Comment:
Could you add a comment to this function explaining what edge case the test function is testing against (why this PR was created) and a small description about how this test function is designed? 

I don't expect anything long.

Reply:
Added [docs](https://github.com/customerio/customerio-android/pull/246/commits/19fce997bd0bd4953128031ebf457ecb515c6578)

Reply:
Great docs. Thanks!